### PR TITLE
Model API implementaion using MetricSink

### PR DESCRIPTION
**Pull request by mwielgus**

### CODE REVIEW

File: api/v1/model_handlers.go [hunk 1]
```diff
@@ -180,53 +185,168 @@ func (a *Api) RegisterModel(container *restful.Container) {
...
 }
 
 // availableMetrics returns a list of available node metric names.
 func (a *Api) availableNodeMetrics(request *restful.Request, response *restful.Response) {
+	a.processMetricNamesRequest(core.NodeKey(request.PathParameter("node-name")), response)
 }
 
 // availableMetrics returns a list of available namespace metric names.
 func (a *Api) availableNamespaceMetrics(request *restful.Request, response *restful.Response) {
+	a.processMetricNamesRequest(core.NodeKey(request.PathParameter("namespace-name")), response)
```
Comment (piosz) [inline api/v1/model_handlers.go]: should be NamespaceKey

File: api/v1/model_handlers.go [hunk 2]
```diff
@@ -180,53 +185,168 @@ func (a *Api) RegisterModel(container *restful.Container) {
...
 
 // nodeMetrics returns a metric timeseries for a metric of the Node entity.
 func (a *Api) nodeMetrics(request *restful.Request, response *restful.Response) {
+	a.processMetricRequest(core.NodeKey(request.PathParameter("node-name")),
+		request, response)
 }
 
 // namespaceMetrics returns a metric timeseries for a metric of the Namespace entity.
 func (a *Api) namespaceMetrics(request *restful.Request, response *restful.Response) {
+	a.processMetricRequest(core.NodeKey(request.PathParameter("namespace-name")),
```
Comment (piosz) [inline api/v1/model_handlers.go]: should be namespaceKey

File: api/v1/model_handlers.go [hunk 3]
```diff
@@ -180,53 +185,168 @@ func (a *Api) RegisterModel(container *restful.Container) {
...
+	a.processMetricRequest(
+		core.PodKey(request.PathParameter("namespace-name"),
+			request.PathParameter("pod-name")),
+		request, response)
 }
 
 func (a *Api) podListMetrics(request *restful.Request, response *restful.Response) {
+	start, end, err := getStartEndTime(request)
+	if err != nil {
+		response.WriteError(http.StatusInternalServerError, err)
```
Comment (piosz) [inline api/v1/model_handlers.go]: bad request

File: api/v1/model_handlers.go [hunk 4]
```diff
@@ -180,53 +185,168 @@ func (a *Api) RegisterModel(container *restful.Container) {
...
+	for _, value := range values {
+		if result.LatestTimestamp.Before(value.Timestamp) {
+			result.LatestTimestamp = value.Timestamp
+		}
+		// TODO: clean up types in model api
+		var intValue int64
+		if value.ValueType == core.ValueInt64 {
+			intValue = value.IntValue
+		} else {
+			intValue = int64(value.FloatValue)
```
Comment (piosz) [inline api/v1/model_handlers.go]: Add support for float in follow-up PR

File: api/v1/model_handlers.go (hunk 1 above)
Comment (mwielgus) [inline api/v1/model_handlers.go]: done

File: api/v1/model_handlers.go (hunk 2 above)
Comment (mwielgus) [inline api/v1/model_handlers.go]: done

File: api/v1/model_handlers.go (hunk 3 above)
Comment (mwielgus) [inline api/v1/model_handlers.go]: done

File: api/v1/model_handlers.go (hunk 4 above)
Comment (mwielgus) [inline api/v1/model_handlers.go]: will do (me or @jszczepkowski)

**Comment by piosz** [comment]
LGTM

**Pull request merged**